test/instance-replacement-tests.js

```diff
@@ -24,6 +24,27 @@ test('Instance replacement', assert => {
   })();
 });
 
+test('Instance not replaced', assert => {
+  const newInstance = undefined;
+
+  compose({
+    initializers: [
+      () => {
+        return newInstance;
```

Comment:
I completely overlooked `const newInstance = undefined;`. Please move it into the initializer function to make it more clear. I don't see any reason to put that in the outer closure scope.